internal/plugintest/util.go

```diff
@@ -1,20 +1,31 @@
 package plugintest
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
 )
 
-func symlinkFile(src string, dest string) (err error) {
-	err = os.Symlink(src, dest)
-	if err == nil {
-		srcInfo, err := os.Stat(src)
-		if err != nil {
-			err = os.Chmod(dest, srcInfo.Mode())
-		}
+func symlinkFile(src string, dest string) error {
+	err := os.Symlink(src, dest)
+
+	if err != nil {
+		return fmt.Errorf("unable to symlink %q to %q: %w", src, dest, err)
 	}
 
-	return
+	srcInfo, err := os.Stat(src)
+
+	if err != nil {
+		return fmt.Errorf("unable to stat %q: %w", src, err)
+	}
+
+	err = os.Chmod(dest, srcInfo.Mode())
+
+	if err != nil {
+		return fmt.Errorf("unable to set %q permissions: %w", dest, err)
+	}
+
+	return nil
```

Comment:
I was trying to read the previous version and getting very confused, until I realised it was starting with a `if err == nil`, the opposite of what I'd have expected.

Now it's WAY more readable and I see even more why you liked that article about "keep the happy path on level-1" :)